plugins/optimization-detective/class-od-link-collection.php

```diff
@@ -52,7 +57,15 @@ final class OD_Preload_Link_Collection implements Countable {
 	 *
 	 * @throws InvalidArgumentException When invalid arguments are provided.
 	 */
-	public function add_link( array $attributes, ?int $minimum_viewport_width, ?int $maximum_viewport_width ): void {
+	public function add_link( array $attributes, ?int $minimum_viewport_width = null, ?int $maximum_viewport_width = null ): void {
+		if ( ! array_key_exists( 'rel', $attributes ) ) {
+			throw new InvalidArgumentException( esc_html__( 'The rel attribute must be provided.', 'optimization-detective' ) );
+		}
+		if ( 'preload' === $attributes['rel'] && ! array_key_exists( 'as', $attributes ) ) {
+			throw new InvalidArgumentException( esc_html__( 'A preload link must include an as attribute.', 'optimization-detective' ) );
+		} elseif ( 'preconnect' === $attributes['rel'] && ! array_key_exists( 'href', $attributes ) ) {
+			throw new InvalidArgumentException( esc_html__( 'A preconnect link must include an href attribute.', 'optimization-detective' ) );
```

Comment:
Nit, I would use sprintf and placeholders for `preconnect`, `preload`, `as` and `href` here, so that they aren't translated. Then this becomes the same string for both, which is less work for translators too.

Reply:
Done in 7cc5e19a4b7531003fbd416e91771a656b8aec44